payment-handler/payment-instruments.https.html

```diff
@@ -0,0 +1,356 @@
+          {
+            src: '/images/rgrg-256x256.png',
+            sizes: '256x256',
+            type: 'image/png',
+          },
+        ],
+        enabledMethods: ['basic-card'],
+        capabilities: {supportedNetworks: ['visa'], supportedTypes: ['credit']},
+      },
+    );
+    result = await registration.paymentManager.instruments.get(
+      'existing-instrument-key',
+    );
+    assert_equals(result.name, 'Instrument Name v2');
+    assert_equals(result.icons.length, 1);
+    assert_equals(
+      result.icons[0].src,
+      new URL('/images/rgrg-256x256.png', window.location.href).href,
+    );
+    assert_equals(result.icons[0].sizes, '256x256');
+    assert_equals(result.icons[0].type, 'image/png');
+    assert_array_equals(result.enabledMethods, ['basic-card']);
+    assert_array_equals(result.capabilities.supportedNetworks, ['visa']);
+    assert_array_equals(result.capabilities.supportedTypes, ['credit']);
+  }, 'Resetting an existing instrument updates the instrument');
+
+  promise_test(async t => {
+    await registration.paymentManager.instruments.clear();
+    await registration.paymentManager.instruments.set(
+      'existing-instrument-key',
+      {
+        name: 'Instrument Name',
+        icons: [
+          {
+            src: '/images/rgrg-256x256.png',
+            sizes: '256x256',
+            type: 'image/png',
+          },
+        ],
+        enabledMethods: ['basic-card'],
+        capabilities: {supportedNetworks: ['mir'], supportedTypes: ['prepaid']},
+      },
+    );
+    await registration.paymentManager.instruments.clear();
+    const result = await registration.paymentManager.instruments.get(
+      'existing-instrument-key',
+    );
+    assert_equals(result, undefined);
+  }, 'Clearing the instruments');
+
+  promise_test(async t => {
+    await registration.paymentManager.instruments.clear();
+    const setPromise = registration.paymentManager.instruments.set(
+      'instrument-key',
+      {
+        name: 'Instrument Name',
+        icons: [
+          {
+            src: '/images/rgrg-256x256.png',
+            sizes: '256x256',
+            type: 'image/jif',
+          },
+        ],
+        enabledMethods: ['basic-card'],
+      },
+    );
+    return promise_rejects(t, new TypeError(), setPromise);
+  }, 'Cannot register instruments with invalid icon media type image/jif');
+
+  promise_test(async t => {
+    await registration.paymentManager.instruments.clear();
+    const setPromise = registration.paymentManager.instruments.set(
+      'instrument-key',
+      {
+        name: 'Instrument Name',
+        icons: [
+          {
+            src: '/images/rgrg-256x256.png',
+            sizes: '256x256',
+            type: 'image/pn' + 'g'.repeat(1000),
+          },
+        ],
+        enabledMethods: ['basic-card'],
+      },
+    );
+    return promise_rejects(t, new TypeError(), setPromise);
+  }, 'Cannot register instruments with very long icon media type image/pngggggg...');
```

Comment:
This restriction seems odd? is this in the MIME spec?

Reply:
My bad. the description really should be "Don't crash when registering instrument with very long icon media type image/pngggg..."